test/controllers/hostgroups_controller_test.rb

```diff
@@ -141,16 +141,37 @@ def setup_user(operation, type = 'hostgroups')
     assert_template :partial => "common/os_selection/_architecture"
   end
 
-  test "should return the selected puppet classes on environment change" do
-    env = FactoryGirl.create(:environment)
-    klass = FactoryGirl.create(:puppetclass)
-    hg = FactoryGirl.create(:hostgroup, :environment => env)
-    assert_equal 0, hg.puppetclasses.length
-    post :environment_selected, { :id => hg.id,
-                                  :hostgroup => { :name => hg.name,
-                                                  :puppetclass_ids => [klass.id],
-                                                  :environment_id => "" }}, set_session_user
-    assert_equal(1, (assigns(:hostgroup).puppetclasses.length))
+  describe '#environment_selected' do
+    setup do
+      @environment = FactoryGirl.create(:environment)
+      @puppetclass = FactoryGirl.create(:puppetclass)
+      @hostgroup = FactoryGirl.create(:hostgroup, :environment => @environment)
+      @params = {
+        id: @hostgroup.id,
+        hostgroup: {
+          name: @hostgroup.name,
+          environment_id: "",
+          puppetclass_ids: [@puppetclass.id]
+        }
+      }
+    end
+
+    test "should return the selected puppet classes on environment change" do
+      assert_equal 0, @hostgroup.puppetclasses.length
+
+      post :environment_selected, @params, set_session_user
+      assert_equal(1, (assigns(:hostgroup).puppetclasses.length))
```

Comment:
Nitpick, you probably want to check `assert_include @puppetclass, assigns(:hostgroup).puppetclasses` to ensure it's the right class